Firestore/core/src/firebase/firestore/model/field_path.h

```diff
@@ -51,6 +52,25 @@ class FieldPath : public impl::BasePath<FieldPath> {
   explicit FieldPath(SegmentsT&& segments) : BasePath{std::move(segments)} {
   }
 
+  /**
+   * Creates and returns a new path from a dot-separated field-path string,
+   * where path segments are separated by a dot ".".
+   *
+   * PORTING NOTE: We define this on the model class to avoid having a tiny
+   * api::FieldPath wrapper class.
+   */
+  static FieldPath FromDotSeparatedString(absl::string_view path);
+
+  /**
+   * Creates and returns a new path from a set of segments received from the
+   * public API.
+   */
+  static FieldPath FromSegments(SegmentsT&& segments) {
```

Comment:
Should we name this in a way that makes it clearer that this is for taking user input? `FromUserSegments`? `FromPublicSegments`? `FromUntrustedSegments`? `FromApiSegments`?

Also please add a comment pointing at the constructor if you're constructing internally.